# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/wgpu_bindings/src/server.rs
# Commit: b3109ad14685
# Full Hash: b3109ad14685624ada7861083ff5332aec9a7818
# Author: Andy Leiserson <aleiserson@mozilla.com>
# Date: 2025-10-07 17:00 +0000
# Description:
#   Bug 1992842 - Update wgpu to upstream 1072b878 (2025-10-06) r=webgpu-reviewers,supply-chain-reviewers,ErichDonGubler
# ==============================================================================

diff -r 1d3f000cf1f9 -r b3109ad14685 gfx/wgpu_bindings/src/server.rs
--- a/gfx/wgpu_bindings/src/server.rs	Tue Oct 07 16:51:44 2025 +0000
+++ b/gfx/wgpu_bindings/src/server.rs	Tue Oct 07 17:00:40 2025 +0000
@@ -25,6 +25,7 @@
 use std::os::raw::c_char;
 use std::ptr;
 use std::sync::atomic::{AtomicU32, Ordering};
+use std::time::Duration;
 
 #[allow(unused_imports)]
 use std::ffi::CString;
@@ -197,7 +198,10 @@
     force_wait: bool,
 ) {
     let maintain = if force_wait {
-        wgt::PollType::Wait
+        wgt::PollType::Wait {
+            submission_index: None,
+            timeout: Some(Duration::from_secs(60)),
+        }
     } else {
         wgt::PollType::Poll
     };